centos.sh

```diff
@@ -1,83 +1,85 @@
-      echo -e "IPADDR=$3" >> $CONFIGFILE 2>>$DEBUGFILE
+    echo -e "#" >> "$CONFIGFILE" 2>> "$DEBUGFILE"
+    echo -e "# device: $1" >> "$CONFIGFILE" 2>> "$DEBUGFILE"
+    echo -e "DEVICE=$1" >> "$CONFIGFILE" 2>> "$DEBUGFILE"
+    echo -e "BOOTPROTO=none" >> "$CONFIGFILE" 2>> "$DEBUGFILE"
+    echo -e "ONBOOT=yes" >> "$CONFIGFILE" 2>> "$DEBUGFILE"
+
+    if [ -n "$3" ] && [ -n "$4" ] && [ -n "$5" ] && [ -n "$6" ] && [ -n "$7" ]; then
+      echo -e "HWADDR=$upper_mac" >> "$CONFIGFILE" 2>> "$DEBUGFILE"
+      echo -e "IPADDR=$3" >> "$CONFIGFILE" 2>> "$DEBUGFILE"
       if is_private_ip "$3"; then
-        echo -e "NETMASK=$5" >> $CONFIGFILE 2>>$DEBUGFILE
-        echo -e "GATEWAY=$6" >> $CONFIGFILE 2>>$DEBUGFILE
+        echo -e "NETMASK=$5" >> "$CONFIGFILE" 2>> "$DEBUGFILE"
+        echo -e "GATEWAY=$6" >> "$CONFIGFILE" 2>> "$DEBUGFILE"
       else
-        echo -e "NETMASK=255.255.255.255" >> $CONFIGFILE 2>>$DEBUGFILE
-        echo -e "SCOPE=\"peer $6\"" >> $CONFIGFILE 2>>$DEBUGFILE
-
-        echo -e "### Hetzner Online GmbH - installimage" > $ROUTEFILE 2>>$DEBUGFILE
-        echo -e "# routing for eth0" >> $ROUTEFILE 2>>$DEBUGFILE
-        echo -e "ADDRESS0=0.0.0.0" >> $ROUTEFILE 2>>$DEBUGFILE
-        echo -e "NETMASK0=0.0.0.0" >> $ROUTEFILE 2>>$DEBUGFILE
-        echo -e "GATEWAY0=$6" >> $ROUTEFILE 2>>$DEBUGFILE
+        echo -e "NETMASK=255.255.255.255" >> "$CONFIGFILE" 2>> "$DEBUGFILE"
+        echo -e "SCOPE=\"peer $6\"" >> "$CONFIGFILE" 2>> "$DEBUGFILE"
+
+        echo -e "### $COMPANY - installimage" > "$ROUTEFILE" 2>> "$DEBUGFILE"
+        echo -e "# routing for eth0" >> "$ROUTEFILE" 2>> "$DEBUGFILE"
+        echo -e "ADDRESS0=0.0.0.0" >> "$ROUTEFILE" 2>> "$DEBUGFILE"
+        echo -e "NETMASK0=0.0.0.0" >> "$ROUTEFILE" 2>> "$DEBUGFILE"
+        echo -e "GATEWAY0=$6" >> "$ROUTEFILE" 2>> "$DEBUGFILE"
       fi
     fi
 
-    if [ "$8" -a "$9" -a "${10}" ]; then
+    if [ -n "$8" ] && [ -n "$9" ] && [ -n "${10}" ]; then
       debug "setting up ipv6 networking $8/$9 via ${10}"
-      echo -e "NETWORKING_IPV6=yes" >> $NETWORKFILE 2>>$DEBUGFILE
-      echo -e "IPV6INIT=yes" >> $CONFIGFILE 2>>$DEBUGFILE
-      echo -e "IPV6ADDR=$8/$9" >> $CONFIGFILE 2>>$DEBUGFILE
-      echo -e "IPV6_DEFAULTGW=${10}" >> $CONFIGFILE 2>>$DEBUGFILE
-      echo -e "IPV6_DEFAULTDEV=$1" >> $CONFIGFILE 2>>$DEBUGFILE
+      echo -e "NETWORKING_IPV6=yes" >> "$NETWORKFILE" 2>> "$DEBUGFILE"
+      echo -e "IPV6INIT=yes" >> "$CONFIGFILE" 2>> "$DEBUGFILE"
+      echo -e "IPV6ADDR=$8/$9" >> "$CONFIGFILE" 2>> "$DEBUGFILE"
+      echo -e "IPV6_DEFAULTGW=${10}" >> "$CONFIGFILE" 2>> "$DEBUGFILE"
+      echo -e "IPV6_DEFAULTDEV=$1" >> "$CONFIGFILE" 2>> "$DEBUGFILE"
     fi
 
     # set duplex/speed
     if ! isNegotiated && ! isVServer; then
-      echo -e "ETHTOOL_OPTS=\"speed 100 duplex full autoneg off\"" >> $CONFIGFILE 2>>$DEBUGFILE
+      echo -e "ETHTOOL_OPTS=\"speed 100 duplex full autoneg off\"" >> "$CONFIGFILE" 2>> "$DEBUGFILE"
```

Comment:
another hit for printf! or simple '' around the text instead of ". no on wants nested "s